lib/core/src/server/config/styleWebpackRules.js

```diff
@@ -0,0 +1,91 @@
+import getCSSModuleLocalIdent from 'react-dev-utils/getCSSModuleLocalIdent';
+
+// style files regexes
+const cssRegex = /\.css$/;
+const cssModuleRegex = /\.module\.css$/;
+const sassRegex = /\.(scss|sass)$/;
+const sassModuleRegex = /\.module\.(scss|sass)$/;
+
+// common function to get style loaders
+const getStyleLoaders = (cssOptions, preProcessor) => {
+  const loaders = [
+    require.resolve('style-loader'),
+    {
+      loader: require.resolve('css-loader'),
+      options: cssOptions,
+    },
+    {
+      // Options for PostCSS as we reference these options twice
+      // Adds vendor prefixing based on your specified browser support in
+      // package.json
+      loader: require.resolve('postcss-loader'),
+      options: {
+        // Necessary for external CSS imports to work
+        // https://github.com/facebook/create-react-app/issues/2677
+        ident: 'postcss',
+        postcss: {},
+        plugins: () => [
+          require('postcss-flexbugs-fixes'), // eslint-disable-line global-require
+          // eslint-disable-next-line global-require
+          require('postcss-preset-env')({
+            autoprefixer: {
+              flexbox: 'no-2009',
+            },
+            stage: 3,
+          }),
+        ],
+      },
+    },
+  ];
+  if (preProcessor) {
+    loaders.push(require.resolve(preProcessor));
+  }
+  return loaders;
+};
+
+export default [
+  // "postcss" loader applies autoprefixer to our CSS.
+  // "css" loader resolves paths in CSS and adds assets as dependencies.
+  // "style" loader turns CSS into JS modules that inject <style> tags.
+  // In production, we use a plugin to extract that CSS to a file, but
+  // in development "style" loader enables hot editing of CSS.
+  // By default we support CSS Modules with the extension .module.css
+  {
+    test: cssRegex,
+    exclude: cssModuleRegex,
+    use: getStyleLoaders({ importLoaders: 1 }),
+  },
+  // Adds support for CSS Modules (https://github.com/css-modules/css-modules)
+  // using the extension .module.css
+  {
+    test: cssModuleRegex,
+    use: getStyleLoaders({
+      importLoaders: 1,
+      modules: true,
+      getLocalIdent: getCSSModuleLocalIdent,
+    }),
+  },
+  // Opt-in support for SASS (using .scss or .sass extensions).
+  // Chains the sass-loader with the css-loader and the style-loader
+  // to immediately apply all styles to the DOM.
+  // By default we support SASS Modules with the
+  // extensions .module.scss or .module.sass
+  {
+    test: sassRegex,
+    exclude: sassModuleRegex,
+    use: getStyleLoaders({ importLoaders: 2 }, 'sass-loader'),
```

Comment:
Does it mean we need to have a `sass-loader` (and `node-sass` probably) in `@storybook/core` ?

Reply:
node sass is a unmanaged dependency for CRA2 vai sass-loader:
https://github.com/webpack-contrib/sass-loader

So we do not need a dependency on `node-sass`

Reply:
Maybe it's worth adding some error logging here.

Something to the extent of: If you've seen this error, we're assuming that you're using CRA 2.0 and have `node-sass` already installed as a dependency.

Reply:
@kylemh Where would this error message be implemented?

Reply:
I just spent like 30 minutes checking into custom error handling on a per-loader basis, but it looks like there isn't anything of the sort. 🤷‍♂️